lighthouse-core/gather/gatherers/trace-elements.js

```diff
@@ -67,22 +69,23 @@ class TraceElements extends Gatherer {
 
   /**
    * @param {Array<LH.TraceEvent>} mainThreadEvents
-   * @return {Array<number>}
+   * @return {Array<TraceElementData>}
    */
-  static getCLSNodeIdsFromMainThreadEvents(mainThreadEvents) {
-    const clsPerNodeMap = new Map();
-    /** @type {Set<number>} */
-    const clsNodeIds = new Set();
+  static getLayoutShiftElements(mainThreadEvents) {
```

Comment:
Gonna add comments throughout this function as it has grown pretty complicated.

Reply:
Agreed. A high-level comment here explaining the approach would be useful. Explaining that `score` is the additional CLS since the last event, and that we want to distribute that incremental score between all the nodes that have shifted, proportional to the impact region of each shift.